Mathlib/Analysis/CStarAlgebra/Module/Synonym.lean

```diff
@@ -0,0 +1,300 @@
+instance instInhabited [Inhabited E] : Inhabited (WithCStarModule E) := ‹Inhabited E›
+instance instNonempty [Nonempty E] : Nonempty (WithCStarModule E) := ‹Nonempty E›
+instance instUnique [Unique E] : Unique (WithCStarModule E) := ‹Unique E›
+
+variable [AddCommGroup E]
+
+/-! ## `WithCStarModule E` inherits various module-adjacent structures from `E`. -/
+
+instance instAddCommGroup : AddCommGroup (WithCStarModule E) := ‹AddCommGroup E›
+instance instSMul {R : Type*} [SMul R E] : SMul R (WithCStarModule E) := ‹SMul R E›
+instance instModule {R : Type*} [Semiring R] [Module R E] : Module R (WithCStarModule E) :=
+  ‹Module R E›
+
+instance instIsScalarTower [SMul R R'] [SMul R E] [SMul R' E]
+    [IsScalarTower R R' E] : IsScalarTower R R' (WithCStarModule E) :=
+  ‹IsScalarTower R R' E›
+
+instance instSMulCommClass [SMul R E] [SMul R' E] [SMulCommClass R R' E] :
+    SMulCommClass R R' (WithCStarModule E) :=
+  ‹SMulCommClass R R' E›
+
+instance instModuleFinite [Semiring R] [Module R E] [Module.Finite R E] :
+    Module.Finite R (WithCStarModule E) :=
+  ‹Module.Finite R E›
+
+
+section Equiv
+
+variable {R E}
+variable [SMul R E] (c : R) (x y : WithCStarModule E) (x' y' : E)
+
+/-! `WithCStarModule.equiv` preserves the module structure. -/
+
+@[simp]
+theorem equiv_zero : equiv E 0 = 0 :=
+  rfl
+
+@[simp]
+theorem equiv_symm_zero : (equiv E).symm 0 = 0 :=
+  rfl
+
+@[simp]
+theorem equiv_add : equiv E (x + y) = equiv E x + equiv E y :=
+  rfl
+
+@[simp]
+theorem equiv_symm_add :
+    (equiv E).symm (x' + y') = (equiv E).symm x' + (equiv E).symm y' :=
+  rfl
+
+@[simp]
+theorem equiv_sub : equiv E (x - y) = equiv E x - equiv E y :=
+  rfl
+
+@[simp]
+theorem equiv_symm_sub :
+    (equiv E).symm (x' - y') = (equiv E).symm x' - (equiv E).symm y' :=
+  rfl
+
+@[simp]
+theorem equiv_neg : equiv E (-x) = -equiv E x :=
+  rfl
+
+@[simp]
+theorem equiv_symm_neg : (equiv E).symm (-x') = -(equiv E).symm x' :=
+  rfl
+
+@[simp]
+theorem equiv_smul : equiv E (c • x) = c • equiv E x :=
+  rfl
+
+@[simp]
+theorem equiv_symm_smul : (equiv E).symm (c • x') = c • (equiv E).symm x' :=
+  rfl
+
+end Equiv
+
+/-- `WithCStarModule.equiv` as a linear equivalence. -/
+@[simps (config := .asFn)]
+def linearEquiv [Semiring R] [Module R E] : C⋆ᵐᵒᵈ E ≃ₗ[R] E :=
+  { LinearEquiv.refl _ _ with
+    toFun := equiv _
+    invFun := (equiv _).symm }
+
+/-! ## `WithCStarModule E` inherits the uniformity and bornology from `E`. -/
+
+variable {E}
+
+instance [u : UniformSpace E] : UniformSpace (C⋆ᵐᵒᵈ E) := u.comap <| equiv E
+
+instance [Bornology E] : Bornology (C⋆ᵐᵒᵈ E) := Bornology.induced <| equiv E
+
+/-- `WithCStarModule.equiv` as a uniform equivalence between `C⋆ᵐᵒᵈ E` and `E`. -/
+def uniformEquiv [UniformSpace E] : C⋆ᵐᵒᵈ E ≃ᵤ E := equiv E |>.toUniformEquivOfUniformInducing ⟨rfl⟩
+
+instance [UniformSpace E] [CompleteSpace E] : CompleteSpace (C⋆ᵐᵒᵈ E) :=
+  uniformEquiv.completeSpace_iff.mpr inferInstance
+
```

Comment:
I think this is not necessary because we will always be constructing a `NormedAddCommGroup` instance on the `WithCStarModule E` type, and these tools aren't needed to make that happen. And once we get that, we'll get these for free.